### DISCUSSION TITLE
Automatically positioned app when it's first open

### DISCUSSION THREAD

**Discussion by s-en-o** [kasper/phoenix#355] [Q&A] [answered]
Hi all,

I have been trying to figure out how the best way to position the application that I just open. I.E If I open an app, I want the app to be position left half of the monitor.
Looking at the example from the wiki, and using Event callback, either with "AppDidLaunch" or "WindowDidOpen", I can't seems to get the app window for some reason - app.windows() or window.app().

If anyone can share some light will be much appreciate it. 

Cheers,
Seno

UPDATE:

I tried the code below with "Notes" app it works well. It recognise the `app.mainWindow()`.
But for some reason with [Zed code editor](https://zed.dev) it's not detecting any main window from `app.mainWindow()`.

```js
const IDAppLaunch = Event.on('appDidLaunch', (app) => {
    const name = app.name()
    const main = app.mainWindow()

    // 👍🏾 This works 
    if (name === 'Notes' && main) {
        tilePosition('rightHalf')
    }
    // ❌ This is not working 
    if (name === 'Zed' && main) {
        tilePosition('leftHalf')
    }
})
```

**Comment by kasper** [comment, marked as answer]
@s-en-o Hi! You might want to try if adding some delay between the event firing and getting the window would help. Apps behave very differently in terms of having focus or setting the main window.

**Reply by s-en-o** [reply]
Hi @kasper, thanks so much for your reply, that is indeed works. Thanks again.

My final code:

```js
const removeTimeout = (timeoutID) => {
	clearTimeout(timeoutID);
};

const IDAppLaunch = Event.on('appDidLaunch', (app) => {
	const name = app.name();
	console.log('=======> App ', name, ' is launched!');

	const timeoutID = setTimeout(() => {
		if (name === 'Code') {
			tilePosition('leftHalf');
		}

		// Clearing timeout
		removeTimeout(timeoutID);
	}, 250); // 250 ms
});
```
